src/Build/Evaluation/Evaluator.cs

```diff
@@ -962,9 +969,10 @@ private void PerformDepthFirstPass(ProjectRootElement currentProjectOrImport)
             if (!Traits.Instance.EscapeHatches.IgnoreTreatAsLocalProperty)
             {
                 IList<string> globalPropertiesToTreatAsLocals = _expander.ExpandIntoStringListLeaveEscaped(currentProjectOrImport.TreatAsLocalProperty, ExpanderOptions.ExpandProperties, currentProjectOrImport.TreatAsLocalPropertyLocation);
-
-                foreach (string propertyName in globalPropertiesToTreatAsLocals)
+                var globalPropertiesToTreatAsLocalsCount = globalPropertiesToTreatAsLocals.Count;
```

Comment:
Yeah via `IList.GetEnumerator`; solving IEnumerable boxing would probably solve 75% of GC problems in .NET

Box `foreach` to stack or something...